homeassistant/components/template/number.py

```diff
@@ -119,10 +120,9 @@ def __init__(
         self._min_value_template = config[ATTR_MIN]
         self._max_value_template = config[ATTR_MAX]
         self._attr_assumed_state = self._optimistic = config[CONF_OPTIMISTIC]
-        self._attr_native_value = None
-        self._attr_native_step = None
-        self._attr_native_min_value = None
-        self._attr_native_max_value = None
+        self._attr_native_step = DEFAULT_STEP
+        self._attr_native_min_value = DEFAULT_MIN_VALUE
+        self._attr_native_max_value = DEFAULT_MAX_VALUE
```

Comment:
These three are not allowed to be defined as `None` in the base `NumberEntity`, so I have adjusted this to use default values instead.